packages/tracing-processing/src/services/db/dbService.ts

```diff
@@ -23,6 +23,15 @@ export function dbServiceBuilder(db: DB) {
               genericLogger.info(
                 `Get enriched purpose ${record.purpose_id}, for tracingId: ${tracing.tracingId}`,
               );
+
+              if (isPurposeAndStatusNotUnique(record, records)) {
+                return enrichPurposeWithError(
```

Comment:
can't we just throw a custom error with our required fields and leave the `EnrichedPurpose` clean?

Reply:
Changed the signature of the function `getEnrichedPurpose` to return `Promise<(PurposeErrorMessage | EnrichedPurpose)[]>`

EnrichedPurpose now doesn't have errorCode and message on its model.

`processingService` is now safeParsing the returning value of  `getEnrichedPurpose`.